### PR TITLE
#731 Enforce Conventional Commits

### LINKED ISSUE #731: Create git-hook for conventional commits

### PR DESCRIPTION
**Pull request by subzero10**
## Status
**READY**

## Description
Fixes #731 
Adds a git hook to automatically check commit messages and enforce [Conventional Commits](https://www.conventionalcommits.org) specification.
This is how it works:
- [Husky](https://www.npmjs.com/package/husky) is used to setup the git hook. It is initialized once with [npm prepare](https://docs.npmjs.com/cli/v8/using-npm/scripts#life-cycle-scripts)
- The git hook runs [commitlint](https://commitlint.js.org/) to enforce commit convention (configured to use Conventional Commits)

**Tip**: If you are using an IDE, you can also get auto-completion for your commit messages:
- [Jetbrains (Webstorm, Intellij, etc.)](https://plugins.jetbrains.com/plugin/13389-conventional-commit)
- [VS Code](https://marketplace.visualstudio.com/items?itemName=vivaxy.vscode-conventional-commits)

## Todos
- [x] husky to setup git-hooks
- [x] commitlint to enforce commit convention
- [x] Changelog Entry (unreleased)
- [x] Update _Changelog_ section in `README.md`
- [x] shipjs with Conventional Commits (setup but not enable yet)

### CODE REVIEW

**Comment by subzero10** [comment]
@joshuap I guess CC should be enforced on all Honeybadger projects right?

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (shalvah) [inline .husky/commit-msg:1]: Does this script run locally? I'm on Windows, and this wouldn't run.

Comment (subzero10) [reply .husky/commit-msg:1]: 😱  That's a very good point! Let me look into that!

Comment (subzero10) [reply .husky/commit-msg:1]: OK, I think we are good with Windows:
From Husky docs:

> [Does it work on Windows?](https://typicode.github.io/husky/#/?id=does-it-work-on-windows)
> Yes. When you install Git on Windows, it comes with the necessary software to run shell scripts.

**Review by shalvah — DISMISSED** [review]

**Comment by subzero10** [comment]
Hey @shalvah, can you please re-approve, I rebased from master. Thanks!

**Review by shalvah — APPROVED** [review]

**Pull request merged**